src/basic/process-util.c

```diff
@@ -1549,6 +1549,31 @@ _noreturn_ void freeze(void) {
                 pause();
 }
 
+int get_process_threads(pid_t pid) {
+        _cleanup_free_ char *t = NULL;
+        const char *p;
+        int n, r;
+
+        if (pid < 0)
+                return -EINVAL;
+
+        p = procfs_file_alloca(pid, "status");
+
+        r = get_proc_field(p, "Threads", WHITESPACE, &t);
+        if (r == -ENOENT)
+                return proc_mounted() == 0 ? -ENOSYS : -ESRCH;
+        if (r < 0)
+                return r;
+
+        r = safe_atoi(t, &n);
+        if (r < 0)
+                return r;
+        if (n < 0)
```

Comment:
Is it possible `n == 0`?? How about `if (n <= 0)` ?

Reply:
yes, n== 0 is possible for zombie processes. i.e. processes which died and thus have no tasks anymore, but which haven't been reaped by their parent process yet, and thus still exists as a PID and in /proc/.

Reply:
Oh, I did not know that. Interesting. Thank you.